app/forms/vpc-router-route/shared.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import type { RouteDestination, RouteTarget } from '~/api'
+
+// VPCs can not be specified as a destination in custom routers
+// https://github.com/oxidecomputer/omicron/blob/4f27433d1bca57eb02073a4ea1cd14557f70b8c7/nexus/src/app/vpc_router.rs#L363
+const destTypes: Record<Exclude<RouteDestination['type'], 'vpc'>, string> = {
+  ip: 'IP',
+  ip_net: 'IP net',
+  subnet: 'subnet',
```

Comment:
These labels should be capitalized since they appear in the listbox. Also `IP net` should probably be `IP network` like we're doing in the badge in `RouterRoutePage`.

![Screenshot 2024-08-13 at 12 05 34 PM](https://github.com/user-attachments/assets/6e5bb2d1-7abd-48d7-b297-4d4c0f498b89)